plugins/modules/keycloak_client.py

```diff
@@ -758,9 +758,19 @@ def normalise_cr(clientrep, remove_ids=False):
             if remove_ids:
                 mapper.pop('id', None)
 
+            # Convert bool to string
+            if 'config' in mapper:
+                for key, value in mapper['config'].items():
+                    if isinstance(value, bool):
+                        mapper['config'][key] = str(value).lower()
```

Comment:
I might be overthinking this, but you are modifying items of a dict that you are iterating over. I remember this causing problems in some versions of Python back in the day. But assuming that the tests cover this part of the code, it should be alright.

Also, an alternative way that would not incur in the non-existent problem :laughing: (feel free to disregard) to implement this could be:
```suggestion
            if 'config' in mapper:
                bool_items = {k: str(v).lower() for k,v in mapper['config'].items() if isinstance(v, bool)}
                mapper['config'].update(bool_items)
```

Reply:
I think editing an existing value should be ok, but I'm not a Python expert. For me, both options are fine, please pick your preferred one .

Reply:
Modifying dict values is fine, deleting or adding items would cause issues.

Reply:
Thanks for chipping in @apollo13 !
@amPrimeSign so there you go, we're good the way it is now. :-)

Reply:
@russoz Thanks :)